packages/toolkit/src/combineSlices.ts

```diff
@@ -415,38 +415,88 @@ const getReducers = (slices: Array<AnySlice | AnyApi | ReducerMap>) =>
       : Object.entries(sliceOrMap)
   )
 
-const ORIGINAL_STATE = Symbol()
+const IS_REPLACEABLE = Symbol.for('rtk-is-replaceable')
+const ORIGINAL_REDUCER = Symbol.for('rtk-reducer-proxy-original')
+
+const reducerProxyMap = new WeakMap<Reducer, Reducer>()
+
+const makeReducerProxy = <R extends Reducer>(reducer: R): R => {
+  let proxy = reducerProxyMap.get(reducer)
+  if (!proxy) {
+    proxy = new Proxy(reducer, {
+      get: (target, prop, receiver) => {
+        if (prop === IS_REPLACEABLE) return true
+        if (prop === ORIGINAL_REDUCER) return target
+        return Reflect.get(target, prop, receiver)
+      },
+    })
+    reducerProxyMap.set(reducer, proxy)
+  }
+  return proxy as R
+}
+
+/**
+ * Marks a slice/api/reducer as replaceable, so injectSlices won't throw an error when a new reducer instance is injected with the same name
+ *
+ * ```ts
+ * injectSlices(markReplaceable(fooSlice), markReplaceable(fooApi), { custom: markReplaceable(fooReducer) })
+ * ```
+ *
+ */
+export const markReplaceable = <Input extends AnySlice | AnyApi | Reducer>(
+  input: Input
+): Input => {
+  if ('reducer' in input) {
+    return {
+      ...input,
+      reducer: makeReducerProxy(input.reducer),
+    }
+  }
+  return makeReducerProxy(input) as Input
+}
+
+const isReplaceable = (reducer: Reducer) => !!(reducer as any)[IS_REPLACEABLE]
```

Comment:
I think you can skip the proxy.

```js
const replaceableReducers = new WeakSet()
const getReducer = input => 'reducer' in input ? input.reducer : input
const markReplaceable = (input) => replaceableReducers.add(getReducer(input))
const isReplaceable = input => replaceableReducers.has(getReducer(input))
```

although I'm not sure if that `markReplaceable` is really the best API here... I don't have a better option, but please keep iterating.